scripts/mysql_system_tables_data.sql

```diff
@@ -26,6 +26,14 @@
 -- a plain character
 SELECT LOWER( REPLACE((SELECT REPLACE(@@hostname,'_','\_')),'%','\%') )INTO @current_hostname;
 
+-- Fill "reverse_db" table with default denies for anyone to
+-- access database 'test' and 'test_%' if "reverse_db" table didn't exist
+CREATE TEMPORARY TABLE tmp_reverse_db LIKE db;
+INSERT INTO tmp_reverse_db VALUES ('%','test','','Y','Y','Y','Y','Y','Y','N','Y','Y','Y','Y','Y','Y','Y','Y','N','N','Y','Y','Y');
+INSERT INTO tmp_reverse_db VALUES ('%','test\_%','','Y','Y','Y','Y','Y','Y','N','Y','Y','Y','Y','Y','Y','Y','Y','N','N','Y','Y','Y');
+INSERT INTO reverse_db SELECT * FROM tmp_reverse_db WHERE @had_db_table=0;
+DROP TABLE tmp_reverse_db;
```

Comment:
Why?

Reply:
This is to populate the reverse privilege table for database privileges with 'Y', indicating that it is a deny for that particular database privilege. This means irrespective of whether this corresponding column in grants 'Y' or 'N', the reverse table's 'Y' would dominate and the user will be denied access for this database level privilege. Need to check if this works as expected by reading it in acl_load.

Reply:
I mean why denying access for test user? It is created such that it is denied to access anything by default.

Reply:
I've modified it now. For eg. database 'test', reverse_db allows create, select, insert and update but denies delete and all other privileges. db allows all the above privileges on the other hand. In this case, if a delete fails, then this means 'reverse' dominates over 'grant'. Similarly, for the other 2 databases test1, test2, I've tried to give opposite privileges in a few places. If it is 'N' in both reverse and grant, still it should be taken as not granted. 
We can now test this behaviour by making acl_load read from 'db' and 'reverse_db' and add the check logic where 'reverse' is checked first for it to dominate. If its 'N' in reverse, then, the value in the 'db' table is taken as is.